core/state_transition.go

```diff
@@ -301,32 +301,30 @@ func (st *stateTransition) checkGasFormula() error {
 	return nil
 }
 
-func (st *stateTransition) collectableNativeBalance(amount *uint256.Int) *uint256.Int {
+func (st *stateTransition) getCollectableNativeBalance(amount *uint256.Int) *uint256.Int {
```

Comment:
The prefix `getXXX` typically implies a read-only function, but in this case, the function modifies state by updating `st.usedSGTBalance` and `st.usedNativeBalance`. A more accurate name might be `collectNativeBalance` to reflect its side effects?